src/Nest/Mapping/AttributeBased/ElasticsearchPropertyAttributeBase.cs

```diff
@@ -18,14 +18,23 @@ public abstract class ElasticsearchPropertyAttributeBase : Attribute, IProperty,
 		bool? IProperty.Store { get; set; }
 		bool? IProperty.DocValues { get; set; }
 		IProperties IProperty.Fields { get; set; }
-		SimilarityOption? IProperty.Similarity { get; set; }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		SimilarityOption? IProperty.Similarity { get { return Self.SimilarityHack?.ToEnum<SimilarityOption>(); } set { Self.SimilarityHack = value.GetStringValue(); } }
+#pragma warning restore CS0618 // Type or member is obsolete
+		string IProperty.SimilarityHack { get; set; }
 		Fields IProperty.CopyTo { get; set; }
 
 		public string Name { get; set; }
 		public bool Ignore { get; set; }
 		public bool DocValues { get { return Self.DocValues.GetValueOrDefault(); } set { Self.DocValues = value; } }
 		public string IndexName { get { return Self.IndexName; } set { Self.IndexName = value; } }
 		public SimilarityOption Similarity { get { return Self.Similarity.GetValueOrDefault(); } set { Self.Similarity = value; } }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+		[Obsolete("This is a temporary binary backwards compatible hack to make sure you can specify named similarities in 2.x, scheduled for removal in 5.0")]
+		public string SimilarityHack { get { return Self.SimilarityHack; } set { Self.SimilarityHack = value; } }
```

Comment:
Sounds too official in my mind, by naming it a hack explicitly we clearly signal our patch is a hack that will get removed as soon as we can.